core/src/test/java/com/sequenceiq/cloudbreak/core/bootstrap/service/container/postgres/PostgresConfigServiceTest.java

```diff
@@ -132,26 +144,40 @@ void decorateServicePillarWithPostgresIfNeededTestCertsWhenSslEnabled(boolean ss
-        when(dbCertificateProvider.getSslCertsFilePath()).thenReturn(SSL_CERTS_FILE_PATH);
         Stack stackView = new Stack();
         stackView.setExternalDatabaseEngineVersion(DBVERSION);
+        Cluster cluster = new Cluster();
+        cluster.setDbSslEnabled(sslEnabledForStack);
+        cluster.setDatabaseServerCrn("crn");
         when(stack.getStack()).thenReturn(stackView);
-
+        when(stack.getCluster()).thenReturn(cluster);
+        if (sslEnabledForStack) {
+            when(dbServerConfigurer.isRemoteDatabaseNeeded(any())).thenReturn(true);
+            when(stack.getCloudPlatform()).thenReturn("AWS");
+            when(clusterService.updateDbSslCert(any(), any())).thenReturn("cert1\ncert2");
+            when(externalDatabaseConfig.isExternalDatabaseSslRotationSupportedFor(any())).thenReturn(true);
+            when(dbCertificateProvider.getSslCertsFilePath()).thenReturn(SSL_CERTS_FILE_PATH);
+            when(dbCertificateProvider.getRelatedSslCerts(stack)).thenReturn(new RedbeamsDbSslDetails(rootCerts, sslEnabledForStack));
+        }
         underTest.decorateServicePillarWithPostgresIfNeeded(servicePillar, stack);
 
-        SaltPillarProperties saltPillarProperties = servicePillar.get(POSTGRES_COMMON);
-        assertThat(saltPillarProperties).isNotNull();
-        assertThat(saltPillarProperties.getPath()).isEqualTo("/postgresql/root-certs.sls");
-
-        Map<String, Object> properties = saltPillarProperties.getProperties();
-        assertThat(properties).isNotNull();
-        assertThat(properties).hasSize(1);
-
-        Map<String, String> rootSslCertsMap = (Map<String, String>) properties.get("postgres_root_certs");
-        assertThat(rootSslCertsMap).isNotNull();
-        assertThat(rootSslCertsMap).containsOnly(entry("ssl_certs", "cert1\ncert2"), entry("ssl_certs_file_path", SSL_CERTS_FILE_PATH),
-                entry("ssl_enabled", String.valueOf(sslEnabledForStack)));
+        if (sslEnabledForStack) {
+            SaltPillarProperties saltPillarProperties = servicePillar.get(POSTGRES_COMMON);
+            assertThat(saltPillarProperties).isNotNull();
+            assertThat(saltPillarProperties.getPath()).isEqualTo("/postgresql/root-certs.sls");
+
+            Map<String, Object> properties = saltPillarProperties.getProperties();
+            assertThat(properties).isNotNull();
+            assertThat(properties).hasSize(1);
+
+            Map<String, String> rootSslCertsMap = (Map<String, String>) properties.get("postgres_root_certs");
+            assertThat(rootSslCertsMap).isNotNull();
+            assertThat(rootSslCertsMap).containsOnly(entry("ssl_certs", "cert1\ncert2"), entry("ssl_certs_file_path", SSL_CERTS_FILE_PATH),
+                    entry("ssl_enabled", String.valueOf(sslEnabledForStack)));
+        } else {
+            SaltPillarProperties saltPillarProperties = servicePillar.get(POSTGRES_COMMON);
+            assertThat(saltPillarProperties).isNull();
+        }
```

Comment:
As per my related comments for `PostgresConfigService`, `sslEnabledForStack` only determines the value of the Salt pillar `ssl_enabled`. This test case represents the scenario when `RedbeamsDbSslDetails` contains a nonempty cert set (`rootCerts` here), so we always expect SSL Salt pillars to be populated.

Therefore, the `else` branch cannot be encountered here, and the `if` can be dropped.